### ISSUE TITLE
cmake error prevents package from building

### ISSUE THREAD

**Issue by fangq** [CGAL/cgal#5857]
I am not entirely sure if this is a cgal's bug or other packages'. My package octave-iso2mesh has trouble building in Fedora rawhide (F35) due to a cmake error, as reported in this bug

https://bugzilla.redhat.com/show_bug.cgi?id=1917433

## Issue Details

octave-iso2mesh can not be built on Fedora F35 but it builds fine on F34 or older releases. The difference in F35 is that cgal was upgraded from 5.2.x to 5.3.x.

here is one of the failed build-log
https://kojipkgs.fedoraproject.org/work/tasks/6035/71816035/build.log

detailed build history can be found in this link
https://koschei.fedoraproject.org/package/octave-iso2mesh

the error were raised in this section of cmake
```
...
CMake Warning (dev) at /usr/share/cmake/CGAL/CGAL_add_test.cmake:78 (if):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.
  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /usr/share/cmake/CGAL/CGAL_add_test.cmake:274 (cgal_add_compilation_test)
  /usr/share/cmake/CGAL/CGAL_CreateSingleSourceCGALProgram.cmake:63 (cgal_add_test)
  CMakeLists.txt:59 (create_single_source_cgal_program)
This warning is for project developers.  Use -Wno-dev to suppress it.
CMake Error at /usr/share/cmake/CGAL/CGAL_add_test.cmake:78 (if):
  if given arguments:
    "TEST" "compilation_of__mesh_polyhedral_domain"
  Unknown arguments specified
Call Stack (most recent call first):
  /usr/share/cmake/CGAL/CGAL_add_test.cmake:274 (cgal_add_compilation_test)
  /usr/share/cmake/CGAL/CGAL_CreateSingleSourceCGALProgram.cmake:63 (cgal_add_test)
  CMakeLists.txt:59 (create_single_source_cgal_program)
-- Configuring incomplete, errors occurred!
```

cmake works fine in F34, for example, in this build log
https://kojipkgs.fedoraproject.org/work/tasks/8570/71778570/build.log

## Source Code

N/A

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Fedora Rawhide - F35
* Compiler: gcc-11.1.1
* Release or debug mode: release
* Specific flags used (if any):
* CGAL version: 5.3-1.fc35, 5.3-0.1.beta1.fc35
* Boost version: 1.75
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by lrineau** [comment]
Hi, I am a CGAL developer, and also packager owner of the CGAL package in Fedora. I have thus the skills to help you. I will deal with that issue...

**Comment by lrineau** [comment]
Hi Qianqian Fang, I have created a PR for your Fedora project, at https://src.fedoraproject.org/rpms/octave-iso2mesh/pull-request/1.

I have also introduced a workaround in CGAL, to deal with the CMake policy `CMP0064`: https://github.com/CGAL/cgal/pull/5859. But with the two patches I have submitted in my Fedora PR, you do not need the workaround within CGAL.

I suggest you update a lot your CMake support in iso2mesh, because for the moment it refers to CMake versions 2.4, 2.6, and 2.8.4. That was probably copied from CMake scripts from very old versions of CGAL.
